icu4j/main/core/src/test/java/com/ibm/icu/dev/test/lang/UnicodeSetTest.java

```diff
@@ -3198,4 +3205,178 @@ public void TestPatternCodePointComplement() {
+
+    @Test
+    public void testIterationMethodsCodepoints() {
+        for (UnicodeSet us : UNICODE_SETS_TO_ITERATE) {
+
+            // Build the reference test result
+            StringJoiner expected = newResult();
+            int i = 0;
+            while (true) {
+                int cp = us.charAt(i++);
+                if (cp == -1) {
+                    break;
+                }
+                expected.add(codePointToString(cp));
+            }
+
+            UnicodeSetIterator usi = new UnicodeSetIterator(us);
+            StringJoiner fromUnicodeSetIterator = newResult();
+            while (usi.next()) {
+                if (usi.codepoint != UnicodeSetIterator.IS_STRING) {
+                    fromUnicodeSetIterator.add(codePointToString(usi.codepoint));
+                }
+            }
+            assertEquals("code points :: UnicodeSetIterator", expected.toString(), fromUnicodeSetIterator.toString());
+
+            StringJoiner fromIterable = newResult();
+            for (Integer cp : us.codePoints()) {
+                fromIterable.add(codePointToString(cp));
+            }
+            assertEquals("code points :: codePoints", expected.toString(), fromIterable.toString());
+
+            StringJoiner fromStream = newResult();
+            us.codePointStream().forEach(cp -> fromStream.add(codePointToString(cp)));
+            assertEquals("code points :: codePointStream", expected.toString(), fromStream.toString());
+        }
+    }
+
+    @Test
+    public void testIterationMethodsRanges() {
+        final UnicodeSet us = populateUnicodeSet();
+
+        // Build the reference test result
+        StringJoiner expected = newResult();
+        for (EntryRange r : us.ranges()) {
+            expected.add(rangeToString(r));
+        }
+
+        UnicodeSetIterator usi = new UnicodeSetIterator(us);
+        StringJoiner fromUnicodeSetIterator = newResult();
+        while (usi.nextRange()) {
+            if (usi.codepoint != UnicodeSetIterator.IS_STRING) {
+                fromUnicodeSetIterator.add(rangeString(usi.codepoint, usi.codepointEnd));
+            }
+        }
+        assertEquals("ranges :: UnicodeSetIterator", expected.toString(), fromUnicodeSetIterator.toString());
+
+        StringJoiner fromStream = newResult();
+        us.rangeStream().forEach(r -> fromStream.add(rangeToString(r)));
+        assertEquals("ranges :: rangeStream", expected.toString(), fromStream.toString());
+    }
+
+    @Test
+    public void testIterationMethodsStrings() {
+        final UnicodeSet us = populateUnicodeSet();
+
+        // Build the reference test result
+        StringJoiner expected = newResult();
+        for (String str : us.strings()) {
+            expected.add(stringToString(str));
+        }
+
+        UnicodeSetIterator usi = new UnicodeSetIterator(us);
+        StringJoiner fromUnicodeSetIterator = newResult();
+        while (usi.nextRange()) {
+            if (usi.codepoint == UnicodeSetIterator.IS_STRING) {
```

Comment:
- start with skipToStrings(), no need to test if it's a string
- nextRange() works, but next() seems more natural